### ISSUE TITLE
Design improvements for the personal access tokens page and flow

### ISSUE THREAD

**Issue by gtsiolis** [gitpod-io/gitpod#15002]
### Problem to solve

Following up from the ongoing efforts to ship the first [MVC](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#0c2e0d6c8a99404b97ccdc206b573d72) iteration of the personal access tokens feature in https://github.com/gitpod-io/gitpod/issues/14280 we had to cut some corners, leading to some **UX debt** around page layout and flow.

This issue acts as a placeholder **follow-up** issue for most UX nitpicks that had slipped between the cracks. Please, append any other UX issues or open separate issues if it helps to keep the change scope small or the discussion separate.

See relevant comments in https://github.com/gitpod-io/gitpod/pull/14934#issuecomment-1326792002, https://github.com/gitpod-io/gitpod/pull/14899#pullrequestreview-1193049003, https://github.com/gitpod-io/gitpod/pull/14936#pullrequestreview-1194114181, etc. Cc @easyCZ 

- [ ] 1. Align tokens more actions menu button to the right, see screenshots below.
- [ ] 2. Use proper padding for the created and regenerated labels, see screenshots below.
- [ ] 3. Additionally, this could ideally use the [`PillLabel`](https://github.com/gitpod-io/gitpod/blob/gt/test-incremental-1/components/dashboard/src/components/PillLabel.tsx) component.
- [ ] 4. On new token creation, submit on enter.
- [ ] 5. On new token creation, remove the trailing empty navigation elements, see screenshot below.
- [ ] 6. On edit token form, remove the trailing empty navigation elements, see screenshot below.
- [ ] 7. On new token creation, limit the error alert width, see screenshot below.
- [ ] 8. On created token, replace the dot with the middot HTML character as used in other product areas.
- [ ] 9. On created token, update the date font weight so that not everything is using a strong font weight.
- [ ] 10. On created token, use the same notation we use for dates as everywhere like _Dec 28, 2022_ not _December 28, 2022_.
- [ ] 11. On the token list, use the same notation we use for dates as everywhere like _Dec 28, 2022_ not Mon Dec 28 2022_.
- [ ] 12. Add regeneration action for the token in the more actions menu.
- [ ] 13. Send email to user when they've successfully created or regenerated a token.
- [ ] 14. Send email to user when a token is expiring soon or has expired.

| Variables ✔️  | Integrations ✔️  | Tokens :X: |
|-|-|-|
| <img width="410" alt="Frame 1146" src="https://user-images.githubusercontent.com/120486/204275485-f25ac3b8-09f6-4612-b34a-d832888323f3.png"> | <img width="410" alt="Frame 1144" src="https://user-images.githubusercontent.com/120486/204275491-0b1ef525-2434-41df-ba83-8a964815b364.png"> | <img width="559" alt="Frame 1143" src="https://user-images.githubusercontent.com/120486/204275495-39b97d3d-bc5b-4950-b52b-dc3c164b9f3f.png"> |

| Label padding ✔️ | Label padding :x: |
|-|-|
| <img width="261" alt="Screenshot 2022-11-28 at 2 19 35 PM" src="https://user-images.githubusercontent.com/120486/204276286-39724504-2d96-421a-b34a-92e1282ae200.png"> | <img width="265" alt="Screenshot 2022-11-28 at 2 18 32 PM" src="https://user-images.githubusercontent.com/120486/204276104-ead6c2de-70ba-4724-8777-81fc05bfc9ef.png"> |

| Trailing element on new token | Trailing element on edit token |
|-|-|
| <img width="342" alt="Screenshot 2022-11-28 at 2 21 58 PM" src="https://user-images.githubusercontent.com/120486/204277131-118993a5-9535-428d-91e1-64772779b1fe.png"> | <img width="463" alt="Screenshot 2022-11-28 at 2 23 31 PM" src="https://user-images.githubusercontent.com/120486/204277135-3ceb3d15-b4b1-431d-8eb9-87db47242c7d.png"> |

| Error alert on new token form |
|-|
| <img width="803" alt="Screenshot 2022-11-28 at 2 25 40 PM" src="https://user-images.githubusercontent.com/120486/204277526-c8001c52-af8e-4fba-9bf3-e5494be9eeb0.png"> |

**Comment by mustard-mh** [comment]
Is there any way that `project` or `prebuild` 's URL did not start with `/t/`? Ask because PAT token edit will be like `/tokens/edit/<pat_id>` and it matches regex `/(t/)?:segment1/:segment2?/:segment3?`

https://github.com/gitpod-io/gitpod/blob/d4a9e560a57adb9d63d168b5f24e167cf69ea4be/components/dashboard/src/Menu.tsx#L60

cc @easyCZ @laushinka

**Comment by jeanp413** [comment]
Is there a template for the email message? Is there an example/library we already use to send emails?

> Send email to user when they've successfully created or regenerated a token.
> Send email to user when a token is expiring soon or has expired.

**Comment by laushinka** [comment]
> Is there a template for the email message? Is there an example/library we already use to send emails?

Hmm, I'm not sure if we've had any conversation about this, or even about whether we want to do this. Maybe @gtsiolis can shed some light?

**Comment by easyCZ** [comment]
@jeanp413 Let's keep the email part of the scope of this issue for now. I'll create a separate ticket for that. 

We'll need a bit more work to figure out the infrastructure to actually do this from the public API.

**Comment by gtsiolis** [comment]
> Is there any way ...

@mustard-mh We added the `/t/` prefix a while ago to avoid having URL conflicts, however, there are a few more edge cases that could be relevant here like https://github.com/gitpod-io/gitpod/issues/13988, https://github.com/gitpod-io/gitpod/issues/11510, https://github.com/gitpod-io/gitpod/issues/5516, etc. See also relevant discussions[[1](https://gitpod.slack.com/archives/C02EN94AEPL/p1666171687000629?thread_ts=1666169937.694359&cid=C02EN94AEPL)][[2](https://gitpod.slack.com/archives/C0463A2KQUU/p1667838068996559?thread_ts=1667834471.387539&cid=C0463A2KQUU)] (internal).

> Is there a template for the email message?

@jeanp413 We're using customer.io for sending all emails when signing up, creating a team, etc. but let's continue this in a separate issue as @easyCZ suggested. Cc @laushinka 

We should probably also document how we're using this tool so that others can be unblocked in using this for developing a product feature. See also [relevant task](https://www.notion.so/gitpod/Customer-io-v2-7431896ad7c04b7a8ad45af2632d41c6) (internal). Cc @chrifro @MichaelAring

**Comment by easyCZ** [comment]
I've moved the email notifications into a follow-up epic - [Personal Access Tokens v1.1](https://github.com/gitpod-io/gitpod/issues/15100) which will contain improvements, feedback, and follow-up issues as move towards a fully stable version of PATs.

Will cross them out from the list here.

**Comment by easyCZ** [comment]
The remaining fixes needed for this Issue are:
 5. On new token creation, remove the trailing empty navigation elements, see screenshot below.
 6. On edit token form, remove the trailing empty navigation elements, see screenshot below.

<img width="199" alt="Screenshot 2022-12-01 at 14 21 59" src="https://user-images.githubusercontent.com/1419286/205063652-11c2b454-874a-486e-879d-50fec9b4f977.png">

@jeanp413 @iQQBot @mustard-mh Are you able to look into this one?

**Comment by mustard-mh** [comment]
https://github.com/gitpod-io/gitpod/issues/15002#issuecomment-1331979553

cc @easyCZ

**Comment by easyCZ** [comment]
> if not, we can change the regex to /t/:segment1/:segment2?/:segment3? to fix issues below

I think this is the way to go. In a way, the `team` segment should really only apply with a fixed prefix (which what the updated regex does)
